src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -4978,332 +4184,304 @@ void CorrectZoneAirTemp(EnergyPlusData &state,
-                          PsyCpAirFnW(ZoneAirHumRat(ZoneNum)) / (TimeStepSys * DataGlobalConstants::SecInHour);
+    Real64 volume = 0.0;
+    if (spaceNum > 0) {
+        volume = state.dataHeatBal->space(spaceNum).Volume;
+    } else {
+        volume = thisZone.Volume;
+    }
+    this->AirPowerCap = volume * thisZone.ZoneVolCapMultpSens *
+                        Psychrometrics::PsyRhoAirFnPbTdbW(state, state.dataEnvrn->OutBaroPress, this->MAT, this->ZoneAirHumRat, RoutineName) *
+                        Psychrometrics::PsyCpAirFnW(this->ZoneAirHumRat) / (state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour);
 
-        AirCap = AIRRAT(ZoneNum);
+    // SpaceHB TODO: For now, room air model is only for zones
+    if (spaceNum == 0) {
+        RoomAirModelManager::ManageAirModel(state, zoneNum);
+    }
 
-        ManageAirModel(state, ZoneNum);
+    // Calculate the various heat balance sums
+    this->calcZoneOrSpaceSums(state, true, zoneNum, spaceNum);
 
-        // Calculate the various heat balance sums
-        CalcZoneSums(state, ZoneNum, SumIntGain, SumHA, SumHATsurf, SumHATref, SumMCp, SumMCpT, SumSysMCp, SumSysMCpT);
+    // Sum all convective internal gains except for people: SumIntGainExceptPeople
+    if (state.dataHybridModel->FlagHybridModel_PC) {
+        // TODO: For now, don't do space heat balance with hybrid model
+        this->SumIntGainExceptPeople = InternalHeatGains::SumAllInternalConvectionGainsExceptPeople(state, zoneNum);
+    }
 
-        // Sum all convective internal gains except for people: SumIntGainExceptPeople
-        if (state.dataHybridModel->FlagHybridModel_PC) {
-            SumIntGainExceptPeople = SumAllInternalConvectionGainsExceptPeople(state, ZoneNum);
-        }
+    //    ZoneTempHistoryTerm = (3.0D0 * ZTM1(zoneNum) - (3.0D0/2.0D0) * ZTM2(zoneNum) + (1.0D0/3.0D0) * ZTM3(zoneNum))
+    int ZoneNodeNum = 0;
+    if (spaceNum == 0) {
+        ZoneNodeNum = thisZone.SystemZoneNodeNumber;
+    } else {
+        ZoneNodeNum = state.dataHeatBal->space(spaceNum).SystemZoneNodeNumber;
+    }
 
-        //    ZoneTempHistoryTerm = (3.0D0 * ZTM1(ZoneNum) - (3.0D0/2.0D0) * ZTM2(ZoneNum) + (1.0D0/3.0D0) * ZTM3(ZoneNum))
-        ZoneNodeNum = Zone(ZoneNum).SystemZoneNodeNumber;
+    Real64 SNLoad = 0.0;
 
-        SNLoad = 0.0;
+    if (ZoneNodeNum > 0) { // This zone is controlled by a zone equipment configuration or zone plenum
+        auto &thisSystemNode = state.dataLoopNodes->Node(ZoneNodeNum);
 
-        if (ZoneNodeNum > 0) { // This zone is controlled by a zone equipment configuration or zone plenum
+        // Heat balance coefficients for controlled zone, i.e. with system air flow
+        this->TempDepCoef = this->SumHA + this->SumMCp + this->SumSysMCp;
+        this->TempIndCoef = this->SumIntGain + this->SumHATsurf - this->SumHATref + this->SumMCpT + this->SumSysMCpT +
+                            (this->NonAirSystemResponse / ZoneMult + this->SysDepZoneLoadsLagged);
```

Comment:
The goal was to have functions calculating only on member data, but that's not possible everywhere.